### PR TITLE
[Routing] Remove deprecation layer

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Routing/Annotation/Route.php [hunk 1]
```diff
@@ -24,133 +24,59 @@
...
-        string $name = null,
-        array $requirements = [],
-        array $options = [],
-        array $defaults = [],
-        string $host = null,
-        $methods = [],
-        $schemes = [],
-        string $condition = null,
-        int $priority = null,
+        string | array | null $path = null,
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Annotation/Route.php:38]: I'd personally prefer no spaces around the `|`. Whatever we decide for this, we need a php-cs-fixer rule for that.

Comment (derrabus) [reply src/Symfony/Component/Routing/Annotation/Route.php:38]: I agree with your preference. Currently, PHP CS Fixer (and thus Fabbot) enforces the spaces.

Comment (nicolas-grekas) [reply src/Symfony/Component/Routing/Annotation/Route.php:38]: Reported at https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5733

File: src/Symfony/Component/Routing/Annotation/Route.php [hunk 2]
```diff
@@ -220,27 +146,27 @@ public function getDefaults()
         return $this->defaults;
     }
 
-    public function setSchemes($schemes)
+    public function setSchemes(array | string $schemes)
     {
-        $this->schemes = \is_array($schemes) ? $schemes : [$schemes];
+        $this->schemes = (array) $schemes;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Annotation/Route.php:151]: for a lower branch?

File: src/Symfony/Component/Routing/Annotation/Route.php [hunk 3]
```diff
@@ -24,133 +24,59 @@
...
-                throw new \BadMethodCallException(sprintf('Unknown property "%s" on annotation "%s".', $key, static::class));
-            }
-            $this->$method($value);
+        if (null !== $stateless) {
+            $this->defaults['_stateless'] = $stateless;
         }
     }
 
-    public function setPath($path)
+    public function setPath(string $path)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Annotation/Route.php:79]: I can't remember, why didn't we add these in 5.*

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

File: src/Symfony/Component/Routing/Annotation/Route.php (hunk 3 above)
Comment (derrabus) [reply src/Symfony/Component/Routing/Annotation/Route.php:79]: Me neither, but I felt we should do this now.

**Pull request merged**